internal/resources/slo/resource_slo.go

```diff
@@ -338,6 +365,40 @@ func packQuery(query map[string]interface{}) (gapi.Query, error) {
 		return sloQuery, nil
 	}
 
+	if query["type"] == "ratio" {
+		ratioquery := query["ratio"].([]interface{})[0].(map[string]interface{})
```

Comment:
is this type assertion safe? It would panic if `query["ratio"]` held something besides a list...

Reply:
We are expecting value within the `query["ratio"]` field to be in a list format.